pkg/lightning/backend/local/job_worker_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/pingcap/tidb/pkg/lightning/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegionJobBaseWorker(t *testing.T) {
```

Comment:
we can also check invariants of `job.ref(w.jobWg)` and `job.done(w.jobWg)` in this unit test after a few modification.

Reply:
job worker is not the only one mutating fields of job, we have to start the routines for `job retryer` and `receive executed jobs` to keep the invariant

Reply:
i will add some comment to explain it